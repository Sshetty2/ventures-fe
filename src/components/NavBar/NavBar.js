import React from 'react';
import { NavLink } from 'react-router-dom';

export const NavBar = ({ user }) => {
  return (
    <div className='NavBar'>
      <NavLink to='/'><h2 className='ventures-title'>Ventures</h2></NavLink>
      <div className='subtitle'>
        <p className='ventures-slogan'>A place to find and track outdoor events that speak to you.</p>
        {user.name && <p className='user-name'>Welcome {user.name}!</p>}
      </div>
      <NavLink exact to='/' className='home-button' activeClassName='home-active' />
      <NavLink 
        to={user.name ? '/profile/upcoming' : '/login'}
        className='profile-button' 
        activeClassName='profile-active' />
    </div>
  )
}

export default NavBar;