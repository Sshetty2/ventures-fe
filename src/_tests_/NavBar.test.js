import React from 'react';
import { shallow } from 'enzyme';
import NavBar from '../components/NavBar/NavBar';

describe('NavBar', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NavBar />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});