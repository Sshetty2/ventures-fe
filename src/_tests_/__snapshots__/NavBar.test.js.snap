// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the snapshot 1`] = `
<div
  className="NavBar"
>
  <NavLink
    to="/"
  >
    <h2
      className="ventures-title"
    >
      Ventures
    </h2>
  </NavLink>
  <p
    className="ventures-slogan"
  >
    A place to find and track outdoor events that speak to you.
  </p>
  <NavLink
    className="home-button"
    to="/"
  />
  <NavLink
    className="profile-button"
    to="/profile/upcoming"
  />
</div>
`;
